retry.go

```diff
@@ -47,6 +49,42 @@ func X(x int, maxBackoff time.Duration, f func() bool) {
 	}
 }
 
```

Comment:
FYI in those places where we wanted the ability to break a retry loop early, we use this package https://github.com/lytics/lio/blob/711fd56cbc86e32b9cd8dd31a3e7285d45cac333/src/ext_svcs/gcp/metadata.go#L39

Reply:
Do you mean this retry.XWithContext is better or expbackoff is better?